### ISSUE TITLE
gradle-8.12-20241210012659+0000: model and components tasks introduce deprecation warnings

### ISSUE THREAD

**Issue by rpalcolea** [gradle/gradle#31691]
### Current Behavior

when running either:

`./gradlew model` or `./gradlew components` 

The following deprecation is shown:

```
> Task :model
Invocation of Task.project at execution time has been deprecated. This will fail with an error in Gradle 9.0. Consult the upgrading guide for further information: https://docs.gradle.org/8.12-20241210012659+0000/userguide/upgrading_version_7.html#task_project
```

This is coming from

* [ModelReport.java](https://github.com/gradle/gradle/blob/5670f2e30375f94a477e21a5358c86c11aa69e3c/subprojects/diagnostics/src/main/java/org/gradle/api/reporting/model/ModelReport.java#L92)
* [ComponentReport.java](https://github.com/gradle/gradle/blob/5670f2e30375f94a477e21a5358c86c11aa69e3c/subprojects/diagnostics/src/main/java/org/gradle/api/reporting/components/ComponentReport.java#L71)

### Expected Behavior

Ideally, out of the box Gradle tasks should not use deprecated features

### Self-contained Reproducer Project

Given a build script:

```
plugins {
    id 'java-library'
}

repositories {
    mavenCentral()
}

dependencies {
    testImplementation libs.junit.jupiter
    testRuntimeOnly 'org.junit.platform:junit-platform-launcher'
    api libs.commons.math3
    implementation libs.guava
}

java {
    toolchain {
        languageVersion = JavaLanguageVersion.of(21)
    }
}
```

Running  `./gradlew model` or `./gradlew components`  results in deprecations

### Gradle version

8.12-20241210012659+0000

### Build scan URL (optional)

https://scans.gradle.com/s/mh62wnyvs22fi/deprecations?expanded=WyIzIl0&expanded-stacktrace=WyI1Il0

**Comment by abstratt** [comment]
Thanks for the report. These tasks will be deprecated, but for now they should not trigger deprecations themselves. This is being addressed in 8.12 RC2.

**Comment by abstratt** [comment]
8.12 RC2 should no longer issue deprecations for those tasks.
